src/puppetlabs/puppetdb/utils.clj

```diff
@@ -79,17 +79,17 @@
                      (.getPath (apply io/file export-root-dir file-suffix))
                      contents))
 
-(defn assoc-when
-  "Assoc to `m` only when `k` is not already present in `m`"
-  [m & kvs]
+(defmacro assoc-when
+  "Assocs the provided values with the corresponding keys if and only
+  if the key is not already present in map."
+  [map key val & kvs]
```

Comment:
why pull out the `key` and `val` params here? it looks like the `kvs` arg can handle them, without needing to add a special case.